apps/prairielearn/src/schemas/schemas/infoAssessment.json

```diff
@@ -393,87 +415,128 @@
         }
       }
     },
-    "questionAlternative": {
+    "QuestionAlternativeJsonSchema": {
       "type": "object",
+      "additionalProperties": false,
+      "required": ["id"],
       "properties": {
         "comment": {
           "description": "Arbitrary comment for reference purposes.",
-          "anyOf": [{ "type": "string" }, { "type": "array" }, { "type": "object" }]
+          "$ref": "#/definitions/CommentJsonSchema"
         },
         "points": {
-          "$ref": "#/definitions/points"
+          "$ref": "#/definitions/PointsJsonSchema"
         },
         "autoPoints": {
-          "$ref": "#/definitions/points"
+          "$ref": "#/definitions/PointsJsonSchema"
         },
         "maxPoints": {
-          "$ref": "#/definitions/pointsSingle"
+          "description": "A single point value.",
+          "$ref": "#/definitions/PointsSingleJsonSchema"
         },
         "maxAutoPoints": {
-          "$ref": "#/definitions/pointsSingle"
+          "description": "A single point value.",
+          "$ref": "#/definitions/PointsSingleJsonSchema"
         },
         "manualPoints": {
-          "$ref": "#/definitions/pointsSingle"
+          "description": "A single point value.",
+          "$ref": "#/definitions/PointsSingleJsonSchema"
         },
         "id": {
-          "$ref": "#/definitions/questionId"
+          "$ref": "#/definitions/QuestionIdJsonSchema"
         },
         "forceMaxPoints": {
-          "$ref": "#/definitions/forceMaxPoints"
+          "description": "Whether to force this question to be awarded maximum points on a regrade.",
+          "$ref": "#/definitions/ForceMaxPointsJsonSchema"
         },
         "triesPerVariant": {
           "description": "The maximum number of graded submissions allowed for each question instance.",
           "type": "number",
           "default": 1
         },
         "advanceScorePerc": {
-          "$ref": "#/definitions/advanceScorePerc"
+          "description": "Minimum score percentage to unlock access to subsequent questions",
+          "$ref": "#/definitions/AdvanceScorePercJsonSchema"
         },
         "gradeRateMinutes": {
           "description": "Minimum amount of time (in minutes) between graded submissions to the same question.",
           "type": "number",
           "minimum": 0
+        },
+        "canSubmit": {
+          "description": "A list of group role names that can submit questions in this assessment. Only applicable for group assessments.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "uniqueItems": true,
+          "default": []
+        },
+        "canView": {
+          "description": "A list of group role names that can view questions in this assessment. Only applicable for group assessments.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "uniqueItems": true,
+          "default": []
```

Comment:
Were these missing before? Are these actually supported in code?

Reply:
See https://github.com/PrairieLearn/PrairieLearn/pull/11024#discussion_r1937789596.